.github/workflow/unit_test.yml

```diff
@@ -0,0 +1,23 @@
+name: Build
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
```

Comment:
In the other file you have ``workflow_dispatch:``.  
This is useful to have here to allow us to manually run the workflow.